### PR TITLE
Only catch timeout-related errors in DurableCursor.

### PR DESCRIPTION
**Pull request by ssanderson**
Previously, we were catching any subclass of OperationFailure, which includes several non-recoverable errors.

This PR also brings us back to parity with upstream, which includes a fix from @ehebert .

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,24 +16,8 @@
...
 from setuptools import setup, find_packages
 
-LONG_DESCRIPTION = None
-README_MARKDOWN = None
-
-with open('README.md') as markdown_source:
-    README_MARKDOWN = markdown_source.read()
-
-try:
-    import pandoc
```
Comment (richafrank) [inline setup.py]: 👍 Can we remove 'pyandoc' from the `setup_requires` too?

File: mongo_proxy/mongodb_proxy.py [hunk 2]
```diff
@@ -113,7 +118,8 @@ def __str__(self):
     def __repr__(self):
         return self.method.__repr__()
 
-class MongoProxy:
+
+class MongoProxy(object):
```
Comment (richafrank) [inline mongo_proxy/mongodb_proxy.py]: That's good stuff.

File: mongo_proxy/durable_cursor.py [hunk 3]
```diff
@@ -139,18 +139,14 @@ def alive(self):
     def next(self):
         try:
             next_record = self.cursor.next()
-
-        # OperationFailure is raised when an operation fails inside
-        # the remote DB.  This most commonly occurs when our cursor
-        # has been inactive for 10 minutes or more.
-        except OperationFailure as exc:
+        except (ExecutionTimeout, WTimeoutError) as exc:
```
Comment (richafrank) [inline mongo_proxy/durable_cursor.py]: Do we need to catch `CursorNotFound` as well, or no?

File: setup.py (hunk 1 above)
Comment (ssanderson) [inline setup.py]: Yep, will do.

File: mongo_proxy/durable_cursor.py (hunk 3 above)
Comment (ssanderson) [inline mongo_proxy/durable_cursor.py]: Ah, good catch.  Will update.

**Comment by ssanderson** [comment]
@richafrank updated if you want to take another look.

**Comment by richafrank** [comment]
Looks good to me! We should upstream this too.

**Comment by jbredeche** [comment]
you didn't add yourself to the contributors list?

**Comment by ssanderson** [comment]
I'll push a PR upstream once we've got this merged internally in case we need another patch.

**Pull request merged**